select_assignment2_flicks/Model/Movie.swift

```diff
@@ -41,6 +42,39 @@ struct Movie {
     
     var voteAverage: Double = 0
     
+    var myRootRef = Firebase(url: Constants.FIREBASE_URL)
+    
+    func isFavoriteFireBase(onComplete: (list: [String], value: Bool) -> Void) {
+        let deviceRef = myRootRef.childByAppendingPath(UIDevice.currentDevice().identifierForVendor!.UUIDString)
+        
+        let favoritesRef = deviceRef.childByAppendingPath("favorites")
+        
+        favoritesRef.observeEventType(.Value, withBlock: { snapshot in
+            if let favorites = snapshot.value as? [String] {
```

Comment:
nice job handling errors here! I think having either a `guard` or an `else` statement is a little bit clearer here.